src/pages/index.tsx

```diff
@@ -107,6 +114,99 @@ const Home: NextPage = () => {
     void router.push(url);
   };
 
+  if (loading) return <span className="text-white">Loding versions...</span>;
+
+  return (
+    <>
+      <div className="flex items-center gap-6">
+        <div>
+          <p className="text-xl text-white">Current version:</p>
+          <Select onValueChange={(value) => setCurrentVersion(value)}>
+            <SelectTrigger className="w-[180px] text-white">
+              <SelectValue placeholder="Select version" />
+            </SelectTrigger>
+            <SelectContent>{renderSelectContent(versionOptions)}</SelectContent>
+          </Select>
+        </div>
+
+        <div>
+          <p className="text-xl text-white">Upgrade to:</p>
+
+          <Select
+            onValueChange={(value) => setUpgradeVersion(value)}
+            disabled={!!noUpgradeAvailable}
+            defaultValue={
+              noUpgradeAvailable
+                ? undefined
+                : upgradeVersionOptions[0]!.versions[0]
+            }
+          >
+            <SelectTrigger className="w-[180px] text-white">
+              <SelectValue
+                placeholder={
+                  noUpgradeAvailable ? "No upgrade available" : "Select version"
+                }
+              />
+            </SelectTrigger>
+            <SelectContent>
+              {renderSelectContent(upgradeVersionOptions)}
+            </SelectContent>
+          </Select>
+        </div>
+      </div>
+
+      <div className="flex items-center gap-6 text-white">
+        {Object.keys(features).map((feature) => (
+          <div className="flex items-center space-x-4" key={feature}>
+            <Checkbox
+              id={feature}
+              checked={features[feature as keyof typeof features]}
+              onCheckedChange={(value) =>
+                value !== "indeterminate"
+                  ? setFeatures((prev) => ({
+                      ...prev,
+                      [feature]: value,
+                    }))
+                  : null
+              }
+            />
+            <label
+              htmlFor={feature}
+              className="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
+            >
+              {feature}
+            </label>
+          </div>
+        ))}
+      </div>
+
+      <button
+        className="rounded-md bg-[hsl(280,100%,70%)] px-4 py-2 text-lg font-medium text-white disabled:cursor-not-allowed disabled:opacity-70"
+        disabled={!currentVersion || !upgradeVersion}
+        onClick={() => goToDiff()}
+      >
+        Upgrade
+      </button>
+    </>
+  );
+};
```

Comment:
This is a pretty chunky boi. Should we move this component outside of the file into a separate file ?

Reply:
less than 300 lines? tbh it doesn't make sense, there are two components and both are quite readable :shrug: